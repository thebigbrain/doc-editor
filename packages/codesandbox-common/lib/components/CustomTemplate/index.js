"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _Tags = _interopRequireDefault(require("../Tags"));

var _elements = require("../SandboxCard/elements");

var _elements2 = require("./elements");

var _getSandboxName = require("../../utils/get-sandbox-name");

var BANNER = 'https://codesandbox.io/static/img/banner.png';

var SCREENSHOT_API_URL = function SCREENSHOT_API_URL(id) {
  return "https://codesandbox.io/api/v1/sandboxes/".concat(id, "/screenshot.png");
};

var CustomTemplate = function CustomTemplate(_ref) {
  var template = _ref.template,
      onClick = _ref.onClick,
      i = _ref.i;

  if (!template) {
    return _react["default"].createElement(_elements2.MyTemplate, {
      key: i
    }, _react["default"].createElement("img", {
      height: "109px",
      alt: "loading",
      src: BANNER
    }), _react["default"].createElement(_elements2.Border, null), _react["default"].createElement("div", null, _react["default"].createElement(_elements2.TemplateTitle, null, "Loading")));
  }

  var sandbox = template.sandbox;
  var title = (0, _getSandboxName.getSandboxName)(sandbox);
  return _react["default"].createElement(_elements2.MyTemplate, {
    key: i,
    onClick: onClick,
    overlayHeight: 109
  }, _react["default"].createElement("img", {
    height: "109px",
    src: process.env.NODE_ENV === 'development' ? BANNER : SCREENSHOT_API_URL(sandbox.id) || BANNER,
    alt: title
  }), _react["default"].createElement(_elements.Overlay, null, _react["default"].createElement(_elements.SandboxDescription, null, sandbox.description), sandbox.tags && _react["default"].createElement(_Tags["default"], {
    tags: sandbox.tags
  })), _react["default"].createElement(_elements2.Border, {
    color: template.color
  }), _react["default"].createElement("div", null, _react["default"].createElement(_elements2.TemplateTitle, null, title), _react["default"].createElement(_elements2.TemplateSubTitle, null, sandbox.description)));
};

var _default = CustomTemplate;
exports["default"] = _default;