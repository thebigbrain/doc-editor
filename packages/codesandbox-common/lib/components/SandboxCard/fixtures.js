"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.templateOptions = exports.templates = exports.sandboxWithUndefinedScreenshotUrl = exports.sandboxWithNullScreenshotUrl = exports.sandboxWithUndefinedAuthor = exports.sandboxWithNullAuthor = exports.sandboxWithLongDescription = exports.sandboxWithLongTitle = exports.sandboxWithManyTags = exports.popularSandbox = exports.sandbox = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var sandbox = function sandbox() {
  var config = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread({
    id: '1234',
    title: 'Test Sandbox',
    description: 'A test sandbox',
    author: {
      username: 'Test User',
      avatar_url: 'https://placekitten.com/g/200/200'
    },
    tags: ['Tag 1', 'Tag 2', 'Tag 3'],
    template: 'create-react-app-typescript',
    screenshot_url: 'https://placekitten.com/g/1200/300',
    view_count: 100,
    fork_count: 100,
    like_count: 100
  }, config);
};

exports.sandbox = sandbox;

var popularSandbox = function popularSandbox() {
  var config = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return sandbox(_objectSpread({
    view_count: 999999999,
    fork_count: 999999999,
    like_count: 999999999
  }, config));
};

exports.popularSandbox = popularSandbox;

var sandboxWithManyTags = function sandboxWithManyTags() {
  var config = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return sandbox({
    tags: ['tag1', 'tag2', 'tag3', 'tag4', 'tag5', 'tag6', 'tag7', 'tag8', 'tag9', 'tag10', 'tag11', 'tag12', 'tag13', 'tag14', 'tag16', 'tag17', 'tag18', 'tag19', 'tag20', 'tag21', 'tag22', 'tag23', 'tag24', 'tag25', 'tag26', 'tag27', 'tag28', 'tag29', 'tag30', 'tag31', 'tag32', 'tag33', 'tag34', 'tag35', 'tag36', 'tag37', 'tag38', 'tag39', 'tag40', 'tag41', 'tag42', 'tag43', 'tag44', 'tag45', 'tag46', 'tag47', 'tag48', 'tag49']
  });
};

exports.sandboxWithManyTags = sandboxWithManyTags;

var sandboxWithLongTitle = function sandboxWithLongTitle() {
  var config = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return sandbox(_objectSpread({
    title: 'This is a really really really really really really really really really really really really really really long title'
  }, config));
};

exports.sandboxWithLongTitle = sandboxWithLongTitle;

var sandboxWithLongDescription = function sandboxWithLongDescription() {
  var config = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return sandbox(_objectSpread({
    description: 'This is a really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really really long description'
  }, config));
};

exports.sandboxWithLongDescription = sandboxWithLongDescription;

var sandboxWithNullAuthor = function sandboxWithNullAuthor() {
  var config = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return sandbox({
    author: null
  });
};

exports.sandboxWithNullAuthor = sandboxWithNullAuthor;

var sandboxWithUndefinedAuthor = function sandboxWithUndefinedAuthor() {
  var config = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return sandbox({
    author: undefined
  });
};

exports.sandboxWithUndefinedAuthor = sandboxWithUndefinedAuthor;

var sandboxWithNullScreenshotUrl = function sandboxWithNullScreenshotUrl() {
  var config = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return sandbox({
    screenshot_url: null
  });
};

exports.sandboxWithNullScreenshotUrl = sandboxWithNullScreenshotUrl;

var sandboxWithUndefinedScreenshotUrl = function sandboxWithUndefinedScreenshotUrl() {
  var config = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return sandbox({
    screenshot_url: undefined
  });
};

exports.sandboxWithUndefinedScreenshotUrl = sandboxWithUndefinedScreenshotUrl;
var templates = ['adonis', 'create-react-app', 'vue-cli', 'preact-cli', 'svelte', 'create-react-app-typescript', 'angular-cli', 'parcel', 'cxjs', '@dojo/cli-create-app', 'gatsby', 'marko', 'nuxt', 'next', 'reason', 'apollo', 'sapper', 'nest', 'static', 'styleguidist', 'gridsome', 'vuepress', 'mdx-deck', 'quasar', 'unibit'];
exports.templates = templates;
var templateOptions = templates.reduce(function (acc, key) {
  return _objectSpread({}, acc, (0, _defineProperty2["default"])({}, key, key));
}, {});
exports.templateOptions = templateOptions;