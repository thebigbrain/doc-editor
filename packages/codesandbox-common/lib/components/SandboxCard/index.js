"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _repoForked = _interopRequireDefault(require("react-icons/lib/go/repo-forked"));

var _eye = _interopRequireDefault(require("react-icons/lib/go/eye"));

var _heart = _interopRequireDefault(require("react-icons/lib/go/heart"));

var _icons = _interopRequireDefault(require("../../templates/icons"));

var _templates = _interopRequireDefault(require("../../templates"));

var _urlGenerator = require("../../utils/url-generator");

var _keycodes = require("../../utils/keycodes");

var _elements = require("./elements");

var _Tags = _interopRequireDefault(require("../Tags"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var getScreenshot = function getScreenshot(id) {
  return "https://codesandbox.io/api/v1/sandboxes/".concat(id, "/screenshot.png");
};

var kFormatter = function kFormatter(num) {
  if (num > 999999) {
    return (num / 1000000).toFixed(1) + 'M';
  }

  if (num > 999) {
    return (num / 1000).toFixed(1) + 'K';
  }

  return num;
};

var SandboxCard =
/*#__PURE__*/
function (_React$PureComponent) {
  (0, _inherits2["default"])(SandboxCard, _React$PureComponent);

  function SandboxCard() {
    var _this;

    (0, _classCallCheck2["default"])(this, SandboxCard);
    _this = (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf2["default"])(SandboxCard).apply(this, arguments));

    _this.toggleOpen = function () {
      _this.props.selectSandbox(_objectSpread({}, _this.props.sandbox));
    };

    _this.handleKeyUp = function (e) {
      if (e.keyCode === _keycodes.ENTER) {
        _this.toggleOpen();
      }
    };

    return _this;
  }

  (0, _createClass2["default"])(SandboxCard, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          sandbox = _this$props.sandbox,
          small = _this$props.small,
          noMargin = _this$props.noMargin,
          noHeight = _this$props.noHeight,
          _this$props$defaultHe = _this$props.defaultHeight,
          defaultHeight = _this$props$defaultHe === void 0 ? 152 : _this$props$defaultHe;

      if (!sandbox) {
        return _react["default"].createElement(_elements.Container, {
          style: {}
        }, _react["default"].createElement(_elements.SandboxImage, {
          as: "div",
          style: {
            border: 0,
            height: 150
          }
        }), _react["default"].createElement(_elements.SandboxInfo, null));
      }

      var template = (0, _templates["default"])(sandbox.template);
      var Icon = (0, _icons["default"])(sandbox.template);
      return _react["default"].createElement(_elements.Container, {
        noMargin: noMargin,
        small: small,
        style: {},
        onClick: this.toggleOpen,
        role: "button",
        tabIndex: 0,
        onKeyUp: this.handleKeyUp
      }, _react["default"].createElement(_elements.Image, null, _react["default"].createElement(_elements.SandboxImage, {
        alt: sandbox.title,
        src: sandbox.screenshot_url || getScreenshot(sandbox.id),
        color: template.color(),
        style: {
          height: defaultHeight
        }
      }), _react["default"].createElement(_elements.Overlay, null, _react["default"].createElement(_elements.SandboxDescription, null, sandbox.description), sandbox.tags && _react["default"].createElement(_Tags["default"], {
        tags: sandbox.tags
      }))), _react["default"].createElement(_elements.SandboxInfo, {
        noHeight: noHeight
      }, _react["default"].createElement(_elements.SandboxTitle, {
        color: template.color()
      }, sandbox.title), _react["default"].createElement(_elements.TemplateIcon, null, _react["default"].createElement(Icon, {
        width: 16,
        height: 16
      }))), _react["default"].createElement(_elements.SandboxStats, null, _react["default"].createElement(_elements.Stats, null, _react["default"].createElement("li", null, _react["default"].createElement(_eye["default"], null), kFormatter(sandbox.view_count)), _react["default"].createElement("li", null, _react["default"].createElement(_repoForked["default"], null), kFormatter(sandbox.fork_count)), _react["default"].createElement("li", null, _react["default"].createElement(_heart["default"], null), kFormatter(sandbox.like_count))), sandbox.author && _react["default"].createElement("a", {
        href: (0, _urlGenerator.profileUrl)(sandbox.author.username)
      }, _react["default"].createElement(_elements.Avatar, {
        src: sandbox.author.avatar_url
      }))));
    }
  }]);
  return SandboxCard;
}(_react["default"].PureComponent);

exports["default"] = SandboxCard;