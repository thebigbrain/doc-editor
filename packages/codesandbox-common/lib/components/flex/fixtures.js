"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.alignItemsOptions = exports.justifyContentOptions = void 0;
var justifyContentOptions = ['stretch', 'center', 'space-between', 'space-around', 'space-evenly', null];
exports.justifyContentOptions = justifyContentOptions;
var alignItemsOptions = ['stretch', 'center', 'start', 'end', null];
exports.alignItemsOptions = alignItemsOptions;