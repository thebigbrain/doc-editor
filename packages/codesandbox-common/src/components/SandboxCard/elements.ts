import styled, { css } from 'styled-components';

import { UserWithAvatar } from '../UserWithAvatar';

export const BG_COLOR = '#1C2022';
export const BG_HOVER = '#212629';

export const Overlay = styled.div`
  position: absolute;
  background: rgba(28, 32, 34, 0.9);
  top: 0;
  left: 0;
  width: 100%;
  height: calc(100% - 3px);
  padding: 1rem;
  box-sizing: border-box;
  opacity: 0;
  transition: opacity 200ms ease;
`;

export const Container = styled.div<{ small?: boolean; noMargin?: boolean }>`
  transition: 0.3s ease all;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen,
    Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  position: relative;
  flex: 1;
  min-width: 300px;

  flex-grow: 1;
  width: 100%;
  border-radius: 4px;
  overflow: hidden;
  cursor: pointer;

  margin-right: 0.5rem;
  margin-left: 0.5rem;

  background-color: ${BG_COLOR};
  box-shadow: 0 0 0 rgba(0, 0, 0, 0.3);

  ${props =>
    props.small &&
    css`
      min-width: auto;
    `};

  ${props =>
    props.noMargin &&
    css`
      margin: 0;
    `};

  &:hover {
    ${Overlay} {
      opacity: 1;
    }
  }

  &:hover {
    background-color: ${BG_HOVER};
    transform: translateY(-5px);
    box-shadow: 0 8px 4px rgba(0, 0, 0, 0.3);
  }

  &:last-child {
    flex-grow: 0;
    min-width: calc(33% - 1rem);
  }
`;

export const SandboxTitle = styled.h2`
  color: ${props => props.color};
  font-family: 'Poppins', sans-serif;
  font-size: 0.75rem;
  font-weight: 600;
  margin-bottom: 6px;
  margin-top: 0;

  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  max-height: 20px;

  word-break: break-all;
`;

export const SandboxDescription = styled.p`
  font-size: 0.8rem;
  color: ${props => props.theme.lightText};
  font-weight: 500;
  line-height: 1.3;
  margin: 0;
  margin-bottom: 16px;
  font-size: 12px;
`;

export const SandboxImage = styled.img`
  display: block;
  margin-bottom: 0;
  z-index: 0;
  border-bottom: 3px solid ${props => props.color};
  height: auto;
  width: 100%;
  background-color: ${BG_HOVER};
  border-image-width: 0;
`;

export const SandboxInfo = styled.div<{ noHeight?: boolean }>`
  left: -1px;
  right: -1px;
  padding: 0.75rem;
  padding-bottom: 4px;
  z-index: 1;
  display: flex;
  justify-content: space-between;
`;

export const TemplateIcon = styled.div`
  display: flex;
`;

export const Author = styled(UserWithAvatar)`
  font-size: 0.75rem;
  font-weight: 600;
  text-decoration: none;
  color: ${props => props.theme.new.description};
`;

export const Stats = styled.ul`
  list-style: none;
  padding: 0;
  display: flex;
  font-family: 'Open Sans', 'Helvetica Neue', sans-serif;
  font-size: 12px;
  line-height: 16px;
  display: flex;
  align-items: center;

  color: ${props => props.theme.placeholder};

  li:not(:last-child) {
    margin-right: 8px;
  }

  li {
    display: flex;
    align-items: center;

    svg {
      margin-right: 6px;
      width: 16px;
      color: ${props => props.theme.placeholder.darken(0.3)};
    }
  }
`;

export const Avatar = styled.img`
  width: 1rem;
  height: 1rem;
  border-radius: 4px;
`;

export const SandboxStats = styled.div`
  display: flex;
  padding: 0.75rem;
  justify-content: space-between;
`;

export const Image = styled.div`
  position: relative;
  font-size: 10px;
`;
