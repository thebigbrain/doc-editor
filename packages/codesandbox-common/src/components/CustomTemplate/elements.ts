import styled, { css } from 'styled-components';
import { Overlay } from '../SandboxCard/elements';

export const Border = styled.div<{ color?: string }>`
  ${({ color = `none` }) => css`
    position: relative;
    top: -4px;
    width: 100%;
    height: 4px;
    background: ${color};
  `}
`;

export const TemplateTitle = styled.span`
  display: block;
  width: 100%;
  margin: 6px 12px;
  font-family: Poppins, Roboto, sans-serif;
  font-size: 12px;
  font-weight: 500;
  text-align: left;
`;

export const TemplateSubTitle = styled.span`
  ${({ theme }) => css`
    display: block;
    width: 100%;
    height: 16px;
    padding: 0 12px;
    margin-bottom: 6px;
    color: ${theme.placeholder};
    font-size: 12px;
    text-align: left;
    line-height: 16px;
    overflow: hidden;
    text-overflow: ellipsis;
    box-sizing: border-box;
  `}
`;

export const MyTemplate = styled.button<{ overlayHeight?: number }>`
  ${({ theme, overlayHeight }) => css`
    position: relative;
    width: 203.5px;
    min-width: 203.5px;
    padding: 0;
    border: 2px solid ${theme.background5};
    border-radius: 4px;
    background: ${theme.background2};
    color: ${theme.lightText};
    cursor: pointer;
    box-sizing: border-box;
    overflow: hidden;

    &:focus {
      border: 2px solid ${theme.secondary};
      outline: none;
    }

    img {
      display: block;
      max-width: 100%;
    }

    ${Overlay} {
      height: ${overlayHeight}px;
      text-align: left;
    }

    &:hover {
      ${Overlay} {
        opacity: 1;
      }
    }
  `}
`;
