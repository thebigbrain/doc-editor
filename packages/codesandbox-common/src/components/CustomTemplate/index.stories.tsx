import React from 'react';
import { storiesOf } from '@storybook/react';
import { object } from '@storybook/addon-knobs';
import CustomTemplate from '.';
import { sandbox } from '../SandboxCard/fixtures';

const template = (props = null) => ({
  id: '2321',
  color: '#fff',
  sandbox: sandbox(props),
});

const stories = storiesOf('components/CustomTemplate', module);
stories
  .add('Default', () => (
    <CustomTemplate template={object('template', template())} />
  ))
  .add('No Title', () => (
    <CustomTemplate template={object('template', template({ title: null }))} />
  ))
  .add('No Description', () => (
    <CustomTemplate
      template={object('template', template({ description: null }))}
    />
  ))
  .add('No Tags', () => (
    <CustomTemplate template={object('template', template({ tags: [] }))} />
  ));
