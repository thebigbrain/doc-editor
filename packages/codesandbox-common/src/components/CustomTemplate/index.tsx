import React from 'react';
import Tags from '../Tags';
import { Overlay, SandboxDescription } from '../SandboxCard/elements';
import {
  Border,
  TemplateTitle,
  TemplateSubTitle,
  MyTemplate,
} from './elements';
import { Sandbox } from '../../types';
import { getSandboxName } from '../../utils/get-sandbox-name';

interface Props {
  template?: {
    id: string;
    color: string;
    sandbox: Sandbox;
  };
  i: number;
  onClick?: () => void;
}
const BANNER = 'https://codesandbox.io/static/img/banner.png';
const SCREENSHOT_API_URL = (id: string) =>
  `https://codesandbox.io/api/v1/sandboxes/${id}/screenshot.png`;
const CustomTemplate = ({ template, onClick, i }: Props) => {
  if (!template) {
    return (
      <MyTemplate key={i}>
        <img height="109px" alt="loading" src={BANNER} />
        <Border />
        <div>
          <TemplateTitle>Loading</TemplateTitle>
        </div>
      </MyTemplate>
    );
  }

  const { sandbox } = template;
  const title = getSandboxName(sandbox);

  return (
    <MyTemplate key={i} onClick={onClick} overlayHeight={109}>
      <img
        height="109px"
        src={
          process.env.NODE_ENV === 'development'
            ? BANNER
            : SCREENSHOT_API_URL(sandbox.id) || BANNER
        }
        alt={title}
      />
      <Overlay>
        <SandboxDescription>{sandbox.description}</SandboxDescription>
        {sandbox.tags && <Tags tags={sandbox.tags} />}
      </Overlay>
      <Border color={template.color} />
      <div>
        <TemplateTitle>{title}</TemplateTitle>
        <TemplateSubTitle>{sandbox.description}</TemplateSubTitle>
      </div>
    </MyTemplate>
  );
};

export default CustomTemplate;
