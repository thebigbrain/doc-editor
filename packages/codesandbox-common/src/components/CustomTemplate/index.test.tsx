import 'jest-styled-components';
import React from 'react';
import mountWithTheme from '../../test/themeMount';
import CustomTemplate from '.';
import { sandbox } from '../SandboxCard/fixtures';

const template = (props = null) => ({
  id: '2321',
  color: '#fff',
  sandbox: sandbox(props),
});

describe('<ContributorsBadge /> rendering', () => {
  it('Default', () =>
    expect(
      mountWithTheme(<CustomTemplate template={template()} />)
    ).toMatchSnapshot());
  it('No Title', () =>
    expect(
      mountWithTheme(<CustomTemplate template={template({ title: null })} />)
    ).toMatchSnapshot());
  it('No Description', () =>
    expect(
      mountWithTheme(
        <CustomTemplate template={template({ description: null })} />
      )
    ).toMatchSnapshot());
  it('No Tags', () =>
    expect(
      mountWithTheme(<CustomTemplate template={template({ tags: [] })} />)
    ).toMatchSnapshot());
});
