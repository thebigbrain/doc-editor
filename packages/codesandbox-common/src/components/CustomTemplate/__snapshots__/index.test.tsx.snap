// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContributorsBadge /> rendering Default 1`] = `ReactWrapper {}`;

exports[`<ContributorsBadge /> rendering No Description 1`] = `ReactWrapper {}`;

exports[`<ContributorsBadge /> rendering No Tags 1`] = `ReactWrapper {}`;

exports[`<ContributorsBadge /> rendering No Title 1`] = `ReactWrapper {}`;
